test/stdlib/NSObject-hashing.swift

```diff
@@ -12,7 +12,6 @@ class Foo: NSObject {
   }
 
   override func hash(into hasher: inout Hasher) {
-    // expected-error@-1 {{overriding non-open instance method outside of its defining module}}
-    // expected-error@-2 {{overriding declarations in extensions is not supported}}
+    // expected-error@-1 {{`NSObject.hash(into:)` is not overridable; subclasses can customize hashing by overriding the `hash` property}}
```

Comment:
I'm not sure the statement "NSObject.hash(into:)` is not overridable" is true because you can override `hash(into:)` like any other function. e.g. 
```swift
class A: Hashable {
  static func == (_: A, _ : A) -> Bool { false }
  func hash(into hasher: inout Hasher) {}
}

class B: A {
  override func hash(into hasher: inout Hasher) {}
}
```
To me the old first diagnostic is correct problem here seems exactly what is being said "overriding non-open instance method outside of its defining module" that `NSObject.hash(into:)` is defined on stdlib as not open(which means that the method can't be overriden outside the module it is defined. The second one can be a bit confuse to interpret but what I think it meant is that `NSObject.hash(into:)` is defined on an extension, and you can't override that.
```swift
class A {
  static func == (_: A, _ : A) -> Bool { false }
}

extension A: Hashable {
  func hash(into hasher: inout Hasher) {}
}

class B: A {
  override func hash(into hasher: inout Hasher) {} // error: overriding declarations in extensions is not supported
}
```
This is not possible on the language right now because methods defined on extensions always use direct dispatch, so override call are not possible... 

So to me the diagnostics are already correct here. 
Maybe the wording on the second could be improved for something like `overriding methods declared in extensions is not supported` but I think this wouldn't improve much. 

cc @xedin @varungandhi-apple @hborla

Reply:
It's true that the declaration is from extension, but it's not clear by just looking at the code without actually going into the definition files. 

Overriding NSObject.hashValue is handled specifically as well to produce separate warning like this: `'NSObject.hash(into:)' is not overridable; subclasses can customize hashing by overriding the 'hash' property`. 
I think it can be helpful to produce similar warning for `NSObject.hash(into:)` as well because its implementation should be moved to `hash` property as well. WDYT?

Reply:
Ah sorry, I think I see the point now that you want a specific diagnostic for `NSObject.hash` I was thinking of `hash` in general
Yeah, so this should be fine :)